File: samples/akka-sample-cluster-kubernetes-scala/src/main/scala/akka/sample/cluster/kubernetes/DemoHealthCheck.scala
package akka.sample.cluster.kubernetes

import scala.concurrent.Future

import akka.actor.ActorSystem
import org.slf4j.LoggerFactory

// Enabled in application.conf
class DemoHealthCheck(system: ActorSystem) extends (() => Future[Boolean]) {
  private val log = LoggerFactory.getLogger(getClass)

  override def apply(): Future[Boolean] = {
    log.info("DemoHealthCheck called")
    Future.successful(true)
  }
}

File: samples/akka-sample-cluster-kubernetes-java/src/main/java/akka/cluster/bootstrap/demo/DemoHealthCheck.java
package akka.cluster.bootstrap.demo;

import akka.actor.ActorSystem;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;
import java.util.function.Supplier;

public class DemoHealthCheck implements Supplier<CompletionStage<Boolean>> {
  private final Logger log = LoggerFactory.getLogger(getClass());

  public DemoHealthCheck(ActorSystem system) {
  }

  @Override
  public CompletionStage<Boolean> get() {
    log.info("DemoHealthCheck called");
    return CompletableFuture.completedFuture(true);
  }
}
